python3/amplify/src/lambda_function.py

```diff
@@ -0,0 +1,130 @@
+DEFAULT_TYPE = 'logzio_amplify_access_lambda'
+
+# set logger
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def get_amplify_access_log_link(access_logs_starttime, access_logs_endtime):
+    # type: (str,str) -> str
+    try:
+
+        amplify = boto3.client('amplify')
+
+        result = amplify.generate_access_logs(
+            startTime=access_logs_starttime,
+            endTime=access_logs_endtime,
+            domainName=os.getenv('AMPLIFY_DOMAIN'),
+            appId=os.getenv('AMPLIFY_APP_ID')
+        )
+        return result["logUrl"]
+    except Exception as e:
+        logger.warning(
+            f'Error occurred while trying get access url please check your credentials: {e}.')
+
+
+def convert_csv_to_array_of_logs(log_url):
+    json_array = []
+
+    try:
+        response = urllib.request.urlopen(log_url)
+        lines = [l.decode('utf-8') for l in response.readlines()]
+        reader = csv.DictReader(lines, FIELD_NAMES)
+
+        for row in reader:
+            json_array.append(json.dumps(row))
+
+        if len(json_array) > 0:
+            json_array.pop(0)
+    except Exception as e:
+        logger.warning(
+            f'Error occurred while trying convert csv to array of logs: {e}.')
+
+    return json_array
+
+
+def _extract_aws_amplify_logs_data(event):
+    # type: (dict) -> dict
+    event_time = datetime.datetime.strptime(
+        event["time"], "%Y-%m-%dT%H:%M:%SZ")
+    try:
+        access_logs_endtime = event_time
+
+        access_logs_starttime = access_logs_endtime - \
+            datetime.timedelta(minutes=5)
+
+        log_url = get_amplify_access_log_link(
+            access_logs_starttime, access_logs_endtime)
+
+        logs = convert_csv_to_array_of_logs(log_url)
+
+        return logs
+
+    except ValueError as e:
+        logger.error("Got exception while loading json, message: {}".format(e))
+        raise ValueError("Exception: json loads")
+
+
+def _add_timestamp(log):
+    # type: (dict) -> dict
+    if '@timestamp' not in log:
+        timestamp_val = log['date'] + 'T' + log['time'] + 'Z'
+        log['@timestamp'] = timestamp_val
+        del log['date']
+        del log['time']
+    return log
+
+
+def _get_additional_logs_data(log, context):
+    # type: (dict, 'LambdaContext') -> dict
+    try:
+        log['function_version'] = context.function_version
+        log['invoked_function_arn'] = context.invoked_function_arn
+    except KeyError:
+        logger.info(
+            'Failed to find context value. Continue without adding it to the log')
+
+    try:
+        log['type'] = os.environ['TYPE']
+    except KeyError:
+        logger.info("Using default TYPE 'logzio_amplify_access_lambda'.")
```

Comment:
Please use `DEFAULT_TYPE` in the log message instead of explicitly writing the type.